pkg/webui/console/views/device-overview/index.js

```diff
@@ -45,12 +48,27 @@ const m = defineMessages({
   keysNotExposed: 'Keys are not exposed',
 })
 
-@connect(state => ({
-  device: selectSelectedDevice(state),
-}))
+@connect(state => {
+  const device = selectSelectedDevice(state)
+  const asConfig = selectAsConfig()
+  const currentHost = getHostnameFromUrl(asConfig.base_url)
+  const redirect =
+    currentHost !== device.application_server_address ||
+    currentHost !== device.network_server_address ||
+    currentHost !== device.join_server_address
```

Comment:
```suggestion
const isThisCluster = 
  (!('application_server_address' in device) || asHost === device.application_server_address) ||
  (!('network_server_address' in device) || nsHost === device.network_server_address) ||
  (!('join_server_address' in device) || jsHost === device.join_server_address) ||
const shouldRedirect = !isThisCluster
```

It should be noted that AS or NS addresses missing are weird edge cases. Also note @bafonins comment on integrating this into a selector. Also note our (implicit) convention of marking boolean values with `should/may/is/…` prefixes.

@htdvisser do you agree to the above as way to determine devices that are on "Other cluster"?

Reply:
Why OR isn't it supposed to be an AND?

Reply:
According to Hylke's comment, we should regard devices and registered on other clusters when AS NS and JS are on different clusters. Note that the if statement is basically covering the reverse case, checking whether the device is NOT on another cluster. So whenever any of the component hosts is actually on the current host, the device will be regarded as registered on this cluster.

Reply:
I was simply writing:
```
const isOtherCluster =
    asHost !== device.application_server_address &&
    nsHost !== device.network_server_address &&
    jsHost !== device.join_server_address
```
Because if one of them does not exist and therefore the statement evaluates to true, the other statements could evaluate to false (if the cluster is not the same), and isOtherCluster would be false. Which is correct. Also, if the remaining statements both evaluate to true (as not being on the same cluster) isOtherCluster is going to be true. Or is this way ambiguous in what concerns the existence of the address in the device?

Reply:
Your condition would have marked an end device that is not registered on an AS, NS and JS (no `application_server_address`, `network_server_address`, `join_server_address`) as "other cluster".

However according to the discussion on slack, the correct condition should now be "if none of `*_server_address` are on the current host, consider device as 'On different cluster'". So your condition is correct now.